File: core/src/main/java/forpleuvoir/dhwu1a/core/event/bot/BotEventType.java
package forpleuvoir.dhwu1a.core.event.bot;

/**
 * @author forpleuvoir
 * <p>#project_name dhwu1a
 * <p>#package forpleuvoir.dhwu1a.core.event.bot
 * <p>#class_name BotEventType
 * <p>#create_time 2021/7/3 20:28
 */
public enum BotEventType {
    BotOnlineEvent(BotOnlineEvent.class),
    BotOfflineEventActive(BotOfflineEventActive.class),
    BotOfflineEventForce(BotOfflineEventForce.class),
    BotOfflineEventDropped(BotOfflineEventDropped.class),
    BotReloginEvent(BotReloginEvent.class);


    private final Class<? extends BotEvent> clazz;

    BotEventType(Class<? extends BotEvent> clazz) {
        this.clazz = clazz;
    }

    public Class<? extends BotEvent> getClazz() {
        return clazz;
    }

    public static boolean hasKey(String key) {
        var isHas = false;
        for (BotEventType value : BotEventType.values()) {
                isHas = value.name().equals(key);
            if (isHas) break;
        }
        return isHas;
    }
}

File: core/src/main/kotlin/forpleuvoir/dhwu1a/core/event/bot/BotEventType.kt
package forpleuvoir.dhwu1a.core.event.bot

/**
 * @author forpleuvoir
 *
 * #project_name dhwu1a
 *
 * #package forpleuvoir.dhwu1a.core.event.bot
 *
 * #class_name BotEventType
 *
 * #create_time 2021/7/3 20:28
 */
enum class BotEventType(val clazz: Class<out BotEvent?>) {
    BotOnlineEvent(forpleuvoir.dhwu1a.core.event.bot.BotOnlineEvent::class.java),
    BotOfflineEventActive(forpleuvoir.dhwu1a.core.event.bot.BotOfflineEventActive::class.java),
    BotOfflineEventForce(forpleuvoir.dhwu1a.core.event.bot.BotOfflineEventForce::class.java),
    BotOfflineEventDropped(forpleuvoir.dhwu1a.core.event.bot.BotOfflineEventDropped::class.java),
    BotReloginEvent(forpleuvoir.dhwu1a.core.event.bot.BotReloginEvent::class.java);

    companion object {
        fun hasKey(key: String): Boolean {
            var isHas = false
            for (value in values()) {
                isHas = value.name == key
                if (isHas) break
            }
            return isHas
        }
    }
}